azure/iac/storageaccounts.rego

```diff
@@ -726,3 +726,348 @@ storage_account_latest_tls_configured_metadata := {
+}
+
+
+#
+# PR-AZR-ARM-STR-020
+#
+
+default storage_account_require_encryption = null
+
+azure_attribute_absence["storage_account_require_encryption"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.storage/storageaccounts"
+    not resource.properties.encryption.requireInfrastructureEncryption
+}
+
+source_path[{"storage_account_require_encryption":metadata}] {
+    resource := input.resources[i]
+    lower(resource.type) == "microsoft.storage/storageaccounts"
+    not resource.properties.encryption.requireInfrastructureEncryption
+    metadata:= {
+        "resource_path": [["resources",i,"properties","encryption","requireInfrastructureEncryption"]]
+    }
+}
+
+
+azure_issue["storage_account_require_encryption"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.storage/storageaccounts"
+    resource.properties.encryption.requireInfrastructureEncryption != true
+}
+
+
+source_path[{"storage_account_require_encryption":metadata}] {
+    resource := input.resources[i]
+    lower(resource.type) == "microsoft.storage/storageaccounts"
+    resource.properties.encryption.requireInfrastructureEncryption != true
+    metadata:= {
+        "resource_path": [["resources",i,"properties","encryption","requireInfrastructureEncryption"]]
+    }
+}
+
+storage_account_require_encryption {
+    lower(input.resources[_].type) == "microsoft.storage/storageaccounts"
+    not azure_issue["storage_account_require_encryption"]
+    not azure_attribute_absence["storage_account_require_encryption"]
+}
+
+storage_account_require_encryption = false {
+    azure_issue["storage_account_require_encryption"]
+}
+
+storage_account_require_encryption = false {
+    azure_attribute_absence["storage_account_require_encryption"]
+}
+
+storage_account_require_encryption_err = "Storage account encryption scopes currently disabled for double encryption for data at rest" {
+    azure_issue["storage_account_require_encryption"]
+} else = "microsoft.storage/storageaccounts property 'encryption.requireInfrastructureEncryption' need to be exist. Its missing from the resource. Please set the value to 'true' after property addition." {
+    azure_attribute_absence["storage_account_require_encryption"]
+}
+
+
+storage_account_require_encryption_metadata := {
+    "Policy Code": "PR-AZR-ARM-STR-020",
+    "Type": "IaC",
+    "Product": "AZR",
+    "Language": "ARM template",
+    "Policy Title": "Storage accounts should have infrastructure encryption",
+    "Policy Description": "Enable infrastructure encryption for higher level of assurance that the data is secure. When infrastructure encryption is enabled, data in a storage account is encrypted twice.",
+    "Resource Type": "microsoft.storage/storageaccounts",
+    "Policy Help URL": "",
+    "Resource Help URL": "https://docs.microsoft.com/en-us/azure/templates/microsoft.storage/storageaccounts"
+}
+
+#
+# PR-AZR-ARM-STR-021
+#
```

Comment:
@shahinshirvani `PR-AZR-ARM-STR-020` and `PR-AZR-ARM-STR-021` are same rule. remove duplicate one.
See my comment at https://github.com/prancer-io/prancer-compliance-test/issues/343#issuecomment-991630268 and update accordingly.

Reply:
they are different as I understand
the ```PR-AZR-ARM-STR-021``` checks ```requireInfrastructureEncryption``` for this resource ```microsoft.storage/storageaccounts/encryptionscopes``` but ```PR-AZR-ARM-STR-020``` checks this resource ```microsoft.storage/storageaccounts```
both have same policy but different rules